apollo-router/tests/common.rs

```diff
@@ -0,0 +1,128 @@
+                .with_filter(EnvFilter::from_default_env()),
+        );
+
+        tracing::subscriber::set_global_default(subscriber).unwrap();
+        global::set_text_map_propagator(propagator);
+
+        let router_location = if cfg!(windows) {
+            PathBuf::from_iter(["..", "target", "debug", "router.exe"])
+        } else {
+            PathBuf::from_iter(["..", "target", "debug", "router"])
+        };
+
+        Self {
+            router: Command::new(router_location)
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .args([
+                    "--hr",
+                    "--config",
+                    &PathBuf::from_iter(["..", "apollo-router", "src", "testdata"])
+                        .join(config_location)
+                        .to_string_lossy()
+                        .to_string(),
+                    "--supergraph",
+                    &PathBuf::from_iter(["..", "examples", "graphql", "local.graphql"])
+                        .to_string_lossy()
+                        .to_string(),
+                ])
+                .spawn()
+                .expect("Router should start"),
+        }
+    }
+
+    pub async fn run_query(&self) -> String {
+        let client = reqwest::Client::new();
+        let id = Uuid::new_v4().to_string();
+        let span = info_span!("client_request", unit_test = id.as_str());
+        let _span_guard = span.enter();
+
+        for _i in 0..100 {
+            let mut request = Request::builder()
+                .method(Method::POST)
+                .header(CONTENT_TYPE, HeaderValue::from_static("application/json"))
+                .header("apollographql-client-name", "custom_name")
+                .header("apollographql-client-version", "1.0")
+                .uri(Uri::from_static("http://localhost:4000"))
+                .body(r#"{"query":"{topProducts{name}}","variables":{}}"#.into())
+                .unwrap();
+
+            global::get_text_map_propagator(|propagator| {
+                propagator.inject_context(
+                    &span.context(),
+                    &mut opentelemetry_http::HeaderInjector(request.headers_mut()),
+                )
+            });
+            match client.send(request).await {
+                Ok(result) => {
+                    tracing::debug!(
+                        "got {}",
+                        String::from_utf8(result.body().to_vec()).unwrap_or_default()
+                    );
+                    return id;
+                }
+                Err(e) => {
+                    tracing::debug!("query failed: {}", e);
+                }
+            }
+            tokio::time::sleep(Duration::from_millis(100)).await;
+        }
+        panic!("unable to send successful request to router")
+    }
+}
+
+impl Drop for TracingTest {
+    fn drop(&mut self) {
+        self.router.kill().expect("router could not be halted");
+        global::shutdown_tracer_provider();
```

Comment:
This shouldn't be safe to call from here and may cause tests to hang...

Reply:
Removing, but had a look at the implementation it just causes the trace provider to drop. The drop will happen anyway.